genesis/process/shardGenesisBlockCreator.go

```diff
@@ -53,6 +59,59 @@ func CreateShardGenesisBlock(arg ArgsGenesisBlockCreator) (data.HeaderHandler, e
 	return header, nil
 }
 
+func createShardGenesisAfterHardFork(arg ArgsGenesisBlockCreator) (data.HeaderHandler, error) {
+	processors, err := createProcessorsForShard(arg)
+	if err != nil {
+		return nil, err
+	}
+
+	argsPendingTxProcessor := hardForkProcess.ArgsPendingTransactionProcessor{
+		Accounts:         arg.Accounts,
+		TxProcessor:      processors.txProcessor,
+		RwdTxProcessor:   processors.rwdProcessor,
+		ScrTxProcessor:   processors.scrProcessor,
+		PubKeyConv:       arg.PubkeyConv,
+		ShardCoordinator: arg.ShardCoordinator,
+	}
+	pendingTxProcessor, err := hardForkProcess.NewPendingTransactionProcessor(argsPendingTxProcessor)
+	if err != nil {
+		return nil, err
+	}
+
+	argsShardBlockAfterHardFork := hardForkProcess.ArgsNewShardBlockCreatorAfterHardFork{
+		ShardCoordinator:   arg.ShardCoordinator,
+		TxCoordinator:      processors.txCoordinator,
+		PendingTxProcessor: pendingTxProcessor,
+		ImportHandler:      arg.importHandler,
+		Marshalizer:        arg.Marshalizer,
+		Hasher:             arg.Hasher,
+	}
+	shardBlockCreator, err := hardForkProcess.NewShardBlockCreatorAfterHardFork(argsShardBlockAfterHardFork)
+	if err != nil {
+		return nil, err
+	}
+
+	hdrHandler, bodyHandler, err := shardBlockCreator.CreateNewBlock(
+		arg.ChainID,
+		arg.HardForkConfig.StartRound,
+		arg.HardForkConfig.StartNonce,
+		arg.HardForkConfig.StartEpoch,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	errNotCritical := processors.txCoordinator.SaveBlockDataToStorage(bodyHandler.(*block.Body))
```

Comment:
I would extract bodyHandler.(*block.Body) into variable and check it against type assertion before saving it into storage

Reply:
done